### ISSUE TITLE
[C++] invalid stray bracket should be valid, `namespace std` not highlighted in meta.block

### ISSUE THREAD

**Issue by MattDMo** [sublimehq/Packages#767]
Two errors, found in the C++ stdlib (`/usr/include/c++/6/cmath` on Ubuntu, starting on line 77):
![c bracket error](https://cloud.githubusercontent.com/assets/1776358/21615803/0e92684e-d1ad-11e6-8b4f-5fa6b9f1f1e1.png)

1. Seeing as this is the stdlib, I doubt that bracket is actually illegal. Does anyone know why it's marked as such?

2. Within the `extern` block, `namespace std` is not highlighted when it should be. The scope is `source.c++ meta.block.c++`.

Test code:
```c++
extern "C++"
{
namespace std _GLIBCXX_VISIBILITY(default)
{
_GLIBCXX_BEGIN_NAMESPACE_VERSION
}
}
```

**Comment by wbond** [comment]
My guess is that currently after the `namespace` keyword we are expecting an identifier, and the macro is throwing it off.

The C preprocessor makes our life so easy!

**Comment by MattDMo** [comment]
Maybe you should just include the preprocessor in your regex engine :wink:

**Comment by wbond** [comment]
This is fixed by adafe24729ca7c5b5799549365e0e2313af463a4, thanks for the report!
